### ISSUE TITLE
documentation: JPA best practices

### ISSUE THREAD

**Issue by amarinso** [oasp/oasp4j#286]
We could include a best practices section on how to use JPA and maybe tips on concrete implementations. 

(I'm myself not expert on this area so I'm very interesting on other's contributions :-)

**Comment by amarinso** [comment]
To include on this section:
- When to use Criteria API, JQPL, QueryDSL
- What is the max size of result set being returned
- How about handling large objects (images etc) in database? Any streaming support recommended

**Comment by hohwille** [comment]
You should probably first read our existing documentation and then suggest what to enhance :)

https://github.com/oasp/oasp4j/wiki/guide-dataaccess-layer#persistence

**Comment by hohwille** [comment]
No one worked on this for years. However, now I have extended the guide with lots of expert information.
See e.g.
https://github.com/oasp-forge/oasp4j-wiki/wiki/guide-jpa-performance
https://github.com/oasp-forge/oasp4j-wiki/wiki/guide-jpa-query#advanced-queries
etc.
Thanks for the feedback but it would be great if we can come from a "somebody should improve the documentation" to a "I added some missing hints to the documentation". In this case you clearly stated that you do not have the experience so I am still thankful for the suggestion and do not take it personal but we had been drowning in issues like this. This is a community project and everybody with a github account can edit the documentation and add small improvements.
